services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/attachment/taydennys/TaydennysAttachmentControllerITest.kt

```diff
@@ -63,6 +77,61 @@ class TaydennysAttachmentControllerITest(@Autowired override val mockMvc: MockMv
         confirmVerified(attachmentService, authorizer)
     }
 
+    @Nested
+    inner class GetAttachmentContent {
+        @Test
+        fun `when valid request should return attachment file`() {
+            val attachmentId = UUID.fromString("afc778b1-eb7c-4bad-951c-de70e173a757")
+            every { authorizer.authorizeAttachment(DEFAULT_ID, attachmentId, VIEW.name) } returns
+                true
+            every { attachmentService.getContent(attachmentId) } returns
+                AttachmentContent(FILE_NAME_PDF, APPLICATION_PDF_VALUE, DUMMY_DATA)
+
+            getAttachmentContent(attachmentId = attachmentId)
+                .andExpect(status().isOk)
+                .andExpect(
+                    header().string(CONTENT_DISPOSITION, "attachment; filename=$FILE_NAME_PDF")
+                )
+                .andExpect(content().bytes(DUMMY_DATA))
+
+            verifySequence {
+                authorizer.authorizeAttachment(DEFAULT_ID, attachmentId, VIEW.name)
+                attachmentService.getContent(attachmentId)
+            }
+        }
+
+        @Test
+        @WithAnonymousUser
+        fun `unauthorized should return error`() {
+            getAttachmentContent(resultType = APPLICATION_JSON).andExpectError(HAI0001)
```

Comment:
This should check that the status is 401 Unauthorized.

Reply:
`andExpectError` does that.

Reply:
Which is a bit silly and unexpected but I did not change that in this PR. Maybe that function could be renamed as `andExpectUnauthenticadedError` and there could be another `andExpectError` without checking for unauthenticated.

Reply:
I mixed up this function with the `hankeError(HankeError)`, which checks for an error without status.

Yes, this could be renamed `andExpectUnauthorized` and moved outside the attachment package. Maybe into `HankeErrorResultMatcher.kt`. The function is not attachment-related, so it could be used elsewhere as well.